akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -388,4 +405,60 @@ private[akka] class PersistentShard(
     }
   }
 
+  private def retrieveEntityRecoveryConfig(): Config = {
+    val path = settings.tuningParameters.entityRecoveryStrategyConfigPath
+    if (path.isEmpty)
+      ConfigFactory.empty()
+    else
+      context.system.settings.config.getConfig(path)
+  }
+}
+
+abstract class EntityRecoveryStrategy {
+  import ShardRegion.EntityId
+  import akka.actor.ActorContext
+  def recoverEntities(context: ActorContext, entities: Set[EntityId])
```

Comment:
Thank you for the feedback. 

> Another thought is that this whole pluggable strategy mechanism might be complete overkill. Instead we could provide 2 fixed implementations and implement them in the shard actor.

Happy to do this... the reason for initially taking this approach was to make it more extensible so that perhaps an addition implementation could be done for #15797 at a later stage. I've done the refactor locally... it is a lot simpler.. will push if you think it's best approach. Let me know.